File: app/src/main/java/com/example/paritosh/stormy/SabkaAlertDialogFragment.java
package com.example.paritosh.stormy;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;
import android.support.v4.app.DialogFragment;

public class SabkaAlertDialogFragment extends DialogFragment {

    private static final String ARG_TITLE = "ARG_TITLE";
    private static final String ARG_MESSAGE = "ARG_MESSAGE";

    @StringRes private int titleRes;
    @StringRes private int messageRes;

    public static SabkaAlertDialogFragment newInstance(@StringRes int title, @StringRes int message) {
        Bundle args = new Bundle();
        args.putInt(ARG_TITLE, title);
        args.putInt(ARG_MESSAGE, message);
        SabkaAlertDialogFragment fragment = new SabkaAlertDialogFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            titleRes = getArguments().getInt(ARG_TITLE);
            messageRes = getArguments().getInt(ARG_MESSAGE);
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());


        builder.setTitle(titleRes)
                .setMessage(messageRes)
                .setPositiveButton(R.string.error_button_ok_text, null).create();

        return builder.create();
    }
}

File: app/src/main/java/com/example/paritosh/stormy/SabkaAlertDialogFragment.kt
package com.example.paritosh.stormy

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.support.annotation.StringRes
import android.support.v4.app.DialogFragment

class SabkaAlertDialogFragment : DialogFragment() {

    private val titleRes: Int by lazy { arguments!!.getInt(ARG_TITLE) }
    @StringRes private var messageRes: Int? = null

    companion object {
        private const val ARG_TITLE = "ARG_TITLE"
        private const val ARG_MESSAGE = "ARG_MESSAGE"

        fun newInstance(title: Int, message: Int): SabkaAlertDialogFragment {

            val args = Bundle()
            args.putInt(ARG_TITLE, title)
            args.putInt(ARG_MESSAGE, message)
            val fragment = SabkaAlertDialogFragment()
            fragment.arguments = args

            return fragment
        }
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        messageRes = arguments!!.getInt(ARG_MESSAGE)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        return AlertDialog.Builder(activity).setTitle(titleRes)
                .setMessage(messageRes!!)
                .setPositiveButton(R.string.error_button_ok_text, null).create()

    }
}
